Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -223,6 +225,11 @@ public class JavaScriptLifter: ComponentBase, Lifter {
                 let dest = MemberExpression.new() <> input(0) <> "." <> op.propertyName
                 let expr = AssignmentExpression.new() <> dest <> " = " <> input(1)
                 w.emit(expr)
+
+                // Recollect changes AbstractInterpreter cannot handle
```

Comment:
well it is not that simple, because we need to call `matbeUpdateType` with correct arguments which are dependant on operation and therefore easier handled in big switch

Reply:
What I can imagine is we create `maybeRecollectType` function in the same manner as `maybeUpdateType`
But calling `maybeRecollectType` will still happen in big switch

Reply:
I guess what I'm suggesting is an interface such as
```
struct TypeRecollectionAnalyzer {
    func analyze(instr: Instruction) -> [Variable]
}
```

And then, at the end of the big switch in the JSLifter:
```
            if let expression = output {
                ...
            }
            for v in typeRecollectionAnalyzer.analyze(instr) {
                maybeUpdateType(of: v, instr)
            }
```
Since the analyzer now decides when to recollect type information, it probably also should decide what to recollect, so this interface would allow that without adding much complexity (as the lifter only has a single point where it deals with type recollection)

Reply:
yeah so the big switch will be duplicated in analyzer too. I wanted to avoid it, but ok for now it will be simple anyway